app/services/bulk_upload/lettings/year2023/row_parser.rb

```diff
@@ -863,12 +864,20 @@ def validate_all_charges_given
       field_130: "personal service charge",
       field_131: "support charge",
       field_132: "total charge" }.each do |field, charge|
-      if public_send(field.to_sym).blank?
-        errors.add(field, I18n.t("validations.financial.charges.missing_charges", question: charge))
-      end
+      next if public_send(field.to_sym).present?
+
+      errors.add(:field_128, I18n.t("validations.financial.charges.missing_charges", question: charge)) unless field != :field_128 && field_128.blank?
+      errors.add(:field_129, I18n.t("validations.financial.charges.missing_charges", question: charge)) unless field != :field_129 && field_129.blank?
+      errors.add(:field_130, I18n.t("validations.financial.charges.missing_charges", question: charge)) unless field != :field_130 && field_130.blank?
+      errors.add(:field_131, I18n.t("validations.financial.charges.missing_charges", question: charge)) unless field != :field_131 && field_131.blank?
+      errors.add(:field_132, I18n.t("validations.financial.charges.missing_charges", question: charge)) unless field != :field_132 && field_132.blank?
```

Comment:
This was slightly confusing to work out - I think it's meaning that blank fields get the error for their own missing charge and non-blank fields get the errors for all of the missing charges (which will end up with just the first missing one displaying)?

That makes sense, I'm just wondering if there's a clearer way to write it? (But there might not be - happy to leave it if nothing comes to mind)

Reply:
Yup, was thinking this was confusing when I wrote this, so thanks for validation! I've refactored it a bit, let me know if that's at all easier to consume!